src/AzureAuth/CommandMain.cs

```diff
@@ -302,9 +302,34 @@ public int OnExecute()
             // Small bug in Lasso - Add does not accept a null IEnumerable here.
             this.eventData.Add("settings_scopes", this.authSettings.Scopes ?? new List<string>());
 
+            if (!this.IsAUserInteractiveEnv())
+            {
+                this.logger.LogDebug($"Skipping used based authentication as specified by the env variable(s) {EnvVars.DisableUserBasedAuthentication}/{EnvVars.CorextNonInteractive}.");
+                return 0;

```

Comment:
Is `0` the right thing to return? Does skipping interactivity mean `azureauth` is a no-op? 🤔 I thought that the idea was to try any modes that could proceed without interactivity but skip the rest.

Reply:
The original idea was to try silent auth modes if interactivity is disabled. But the objective of the feature changed to disable any authentication that runs on behalf of a user on non-interactive machines as it's not applicable. Since non-interactive machines do not need to fetch the token silently from the cache, we won't be trying that scenario or any other silent modes.

Reply:
Ignore my previous comment. I just understood what you meant.

Reply:
I agree with your question on returning `0` as the error code. I will return `1` to fail it out loud.